### PR TITLE
Test automation of mon failure scenarios with intransit encryption.

### PR DESCRIPTION
**Pull request by paraggit**
Test case to ensure the proper functioning of mon failures with in-transit encryption.

        Steps:
            1. Ensure that in-transit encryption is enabled on the setup.
            2. Scale down two mons.
            3. Restart Mgr pod.
            4. Sleep for 5 seconds.
            5. Scale up mons.
            6. Sleep for 10 seconds.
            7. Wait for mgr pod to move to the running state.
            8. Verify the in-transit encryption configuration after scaling up mons and restarting mgr pod.

### CODE REVIEW

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import run_cmd
+
+from ocs_ci.ocs.cluster import CephCluster
+
+log = logging.getLogger(__name__)
+
+
+@tier4a
+@skipif_ocs_version("<4.13")
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:24]: polarion id mark is missing

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            ignore_errors (bool, optional): Ignore errors. Defaults to False.
+
+        Returns:
+            None
+        """
+        log.info(f"Scaling mon {mon_name} to replica count {replica_count}")
+        run_cmd(
+            f"oc scale --replicas={replica_count} deploy/{mon_name} -n {constants.OPENSHIFT_STORAGE_NAMESPACE}",
+            ignore_error=ignore_error,
+        )
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:57]: this function could be useful outside this test. I suggest placing it in a helpers file

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            4. Sleep for 5 seconds.
+            5. Scale up mons.
+            6. Sleep for 10 seconds.
+            7. Wait for mgr pod to move to the running state.
+            8. Verify the in-transit encryption configuration after scaling up mons and restarting mgr pod.
+        """
+
+        if not get_in_transit_encryption_config_state():
+            if config.ENV_DATA.get("in_transit_encryption"):
+                pytest.fail("In-transit encryption is not enabled on the setup")
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:76]:
```suggestion
                pytest.fail("In-transit encryption is not enabled on the setup while it was supposed to be")
```

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            else:
+                set_in_transit_encryption()
+
+        ceph_obj = CephCluster()
+
+        log.info("Verifying the in-transit encryption is enable on setup.")
+        assert in_transit_encryption_verification()
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:83]:
```suggestion
        assert in_transit_encryption_verification(), "In transit encryption verification failed"
```

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+        # Select Two mons
+        self.mons = ceph_obj.get_mons_from_cluster()[:2]
+
+        # Scale Down Mon Count to replica=0
+        for mon in self.mons:
+            self.scale_ceph_mon(0, mon)
+
+        time.sleep(10)
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:92]: please add a comment with why the sleep is needed

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+        # Restart Mgr pod
+        mgr_pod = ceph_obj.mgrs[0]
+        mgr_pod.delete(wait=True)
+
+        time.sleep(5)
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:98]: same

File: tests/encryption/test_mon_failure_in_intransit_encryption.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+            resource_count=1,
+            timeout=100,
+        ), "Mgr pod did'nt move to Running state after 100 seconds"
+
+        log.info(
+            "Verifying the in-transit encryption config "
+            "After scaling up mon and restarting mgr pod"
+        )
+
+        assert in_transit_encryption_verification()
```
Comment (ebenahar) [inline tests/encryption/test_mon_failure_in_intransit_encryption.py:120]:
```suggestion
        assert in_transit_encryption_verification(), "In transit encryption verification failed"
```

File: tests/encryption/test_mon_failure_in_intransit_encryption.py (hunk 1 above)
Comment (paraggit) [reply tests/encryption/test_mon_failure_in_intransit_encryption.py:23]: Added.

File: tests/encryption/test_mon_failure_in_intransit_encryption.py (hunk 5 above)
Comment (paraggit) [reply tests/encryption/test_mon_failure_in_intransit_encryption.py:81]: Comment Added

File: tests/encryption/test_mon_failure_in_intransit_encryption.py (hunk 6 above)
Comment (paraggit) [reply tests/encryption/test_mon_failure_in_intransit_encryption.py:88]: Done

File: tests/encryption/test_mon_failure_in_intransit_encryption.py (hunk 3 above)
Comment (paraggit) [reply tests/encryption/test_mon_failure_in_intransit_encryption.py]: Done

File: tests/encryption/test_mon_failure_in_intransit_encryption.py (hunk 7 above)
Comment (paraggit) [reply tests/encryption/test_mon_failure_in_intransit_encryption.py]: Done

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/ocs/cluster.py [hunk 8]
```diff
@@ -3048,3 +3048,22 @@ def get_mds_standby_replay_info():
         "mds_daemon": ceph_daemon_name,
         "standby_replay_pod": standby_replay_pod.name,
     }
+
+
+def scale_ceph_mon(replica_count, mon_name, ignore_error=False):
```
[the remark this answers is not in the document]
Comment (paraggit) [reply ocs_ci/ocs/cluster.py]: I replaced it with  existing method 'modify_deployment_replica_count' to scale  up/down deployment.

**Review by AviadP — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**